### PR TITLE
Upgrading to Terraform version 0.12

### PR DESCRIPTION
**Pull request by 5RK7N**
- Upgrading azurerm provider version to the latest version which compatible with Terraform v0.12 
- Cleaning resource's configuration and interpolation syntax

### CODE REVIEW

**Comment by 5RK7N** [comment]
I cannot find any terraform-test image in v12.0 (https://hub.docker.com/r/microsoft/terraform-test/tags). The latest build v0.12.6 failed :(
- https://github.com/Azure/terraform-test
- https://travis-ci.org/Azure/terraform-test/builds/571386943?utm_source=github_status&utm_medium=notification

File: main.tf [hunk 1]
```diff
@@ -1,64 +1,65 @@
...
   }
 
   storage_profile {
-    storage_mb            = "${var.storage_mb}"
-    backup_retention_days = "${var.backup_retention_days}"
-    geo_redundant_backup  = "${var.geo_redundant_backup}"
+    storage_mb            = var.storage_mb
+    backup_retention_days = var.backup_retention_days
+    geo_redundant_backup  = var.geo_redundant_backup
   }
```
[the remark this answers is not in the document]
Comment (yupwei68) [reply main.tf:21]: `geo_redundant_backup` change to `geo_redundant_backup_enabled`

File: test/fixture/variables.tf [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
 }
 
-variable "fw_rule_prefix" {}
+variable "fw_rule_prefix" {
+}
 
-variable fw_rules {
-  type = "list"
+variable "fw_rules" {
+  type = list(string)
```
Comment (yupwei68) [inline test/fixture/variables.tf:12]: list(any)

**Comment by yupwei68** [comment]
@5RK7N Thanks for your contribution to postgresql module 🚀 . Since you might be unavailable for the changes in comments, I have upgrade the module to Terraform 0.12 based on you work. Thanks again.

**Pull request closed without merging**